### ISSUE TITLE
PKCS12 format is not supported by the PyCrypto library

### ISSUE THREAD

**Issue by hello-josh** [google/identity-toolkit-python-client#16]
I am getting the following error while trying to execute methods on the `GitkitClient`:

``` python
def get_client():
    with open(current_app.config['GOOGLE_PRIVATE_KEY_FILE'], 'rb') as key_file:
        key = key_file.read()
    return gitkitclient.GitkitClient(client_id=current_app.config['GOOGLE_CLIENT_ID'],
                                     service_account_email=current_app.config['GOOGLE_SERVICE_ACCOUNT_EMAIL'],
                                     service_account_key=key,
                                     widget_url=url_for('widget', _external=True),  # flask route for the widget
                                     cookie_name=current_app.config['GITKIT_COOKIE_NAME'])

get_client().GetClientId()  # throws
```

> NotImplementedError: PKCS12 format is not supported by the PyCrypto library. Try converting to a "PEM" (openssl pkcs12 -in xxxxx.p12 -nodes -nocerts > privatekey.pem) or using PyOpenSSL if native code is an option.

`GitkitClient.VerifyGitkitToken` works as expected.

**Comment by liujin-google** [comment]
You can convert the .p12 file to .pem using the command in the error message, and set GOOGLE_PRIVATE_KEY_FILE to be the full file path of the pem file.

**Comment by hello-josh** [comment]
Right, the error tells me to convert the file to a `.pem' but the README says to use a`.p12` file.

Is this a bug? A documentation error?

Has anyone else does this on Google App Engine without converting to `.pem`?

**Comment by dunkatron** [comment]
I dealt with this today. The documentation is incorrect. If you look at the code it clearly expects an unencrypted PEM file containing your private key. There's no way to export a file in this format from the Appengine Console. You have to export a P12 file, then use OpenSSL command line tools to extract the key from it in unencrypted PEM format.

**Comment by dunkatron** [comment]
The documentation was clearly written without looking at the code. Extremely disappointing from a product that purports to alleviate all of the pains of SSO.
